packages/css-data/src/property-parsers/filter.ts

```diff
@@ -0,0 +1,63 @@
+import * as csstree from "css-tree";
+import type {
+  InvalidValue,
+  TupleValue,
+  TupleValueItem,
+} from "@webstudio-is/css-engine";
+import { cssTreeTryParseValue, isValidDeclaration } from "../parse-css-value";
+
+/*
+  https://github.com/webstudio-is/webstudio/issues/1016
+  https://github.com/csstree/csstree/issues/246
+
+  css-tree can't validate filter values.
+  So, we are relying on the isValidDeclaration function to validate the filter value.
+  Which uses browser CSSStyleValue.parse to validate.
+*/
+
+export const parseFilter = (input: string): TupleValue | InvalidValue => {
+  let tokenStream = input.trim();
+  tokenStream = tokenStream.endsWith(";")
+    ? tokenStream.slice(0, -1)
+    : tokenStream;
+
+  const cleanupKeywords = ["filter:"];
+
+  for (const cleanupKeyword of cleanupKeywords) {
+    tokenStream = tokenStream.startsWith(cleanupKeyword)
+      ? tokenStream.slice(cleanupKeyword.length).trim()
+      : tokenStream;
+  }
+
+  const cssAst = cssTreeTryParseValue(tokenStream);
+  if (cssAst === undefined) {
+    return {
+      type: "invalid",
+      value: input,
+    };
+  }
+
+  const isValidFilterDecleration = isValidDeclaration("filter", input);
+  if (isValidFilterDecleration === false) {
+    return {
+      type: "invalid",
+      value: input,
+    };
+  }
+
+  const layers: TupleValueItem[] = [];
+  csstree.walk(cssAst, (node) => {
+    if (node.type === "Value") {
+      for (const child of node.children) {
+        if (child.type === "Function") {
+          layers.push({ type: "keyword", value: csstree.generate(child) });
```

Comment:
In no case it's keyword we have UnparsedValue for such things from what I know

Reply:
shouldn't we have type: "function"?

Reply:
so far the only function we had was rgb() and we added rgb as a type, not as a generic function, maybe it makes sense to just keep adding specific functions? since you want to make sure each function interface is strictly defined